javaparser-core/src/main/java/com/github/javaparser/printer/lexicalpreservation/Difference.java

```diff
@@ -175,129 +143,233 @@ void apply() {
+        if (diffIndex < diffElements.size() && originalIndex >= originalElements.size()) {
+            DifferenceElement diffElement = diffElements.get(diffIndex);
+            if (diffElement instanceof Kept) {
+                Kept kept = (Kept) diffElement;
+
+                if (kept.isWhiteSpaceOrComment() || kept.isIndent() || kept.isUnindent()) {
+                    diffIndex++;
+                } else {
+                    throw new IllegalStateException("Cannot keep element because we reached the end of nodetext: "
+                            + nodeText + ". Difference: " + this);
+                }
+            } else if (diffElement instanceof Added) {
+                Added addedElement = (Added) diffElement;
 
-        Map<Integer, Integer> nodeTextIndexToPreviousCSMIndex = new HashMap<>();
-        for (int i=0;i<nodeTextIndexOfPreviousElements.size();i++) {
-            int value = nodeTextIndexOfPreviousElements.get(i);
-            if (value != -1) {
-                nodeTextIndexToPreviousCSMIndex.put(value, i);
+                nodeText.addElement(originalIndex, addedElement.toTextElement());
+                originalIndex++;
+                diffIndex++;
+            } else {
+                throw new UnsupportedOperationException(diffElement.getClass().getSimpleName());
             }
+
+            isLeftOverElement = true;
         }
-        int lastNodeTextIndex = nodeTextIndexOfPreviousElements.stream().max(Integer::compareTo).orElse(-1);
 
-        // Elements to be added at the end
-        List<CsmElement> elementsToBeAddedAtTheEnd = new LinkedList<>();
-        List<CsmElement> nextOrderElements = elementsFromNextOrder.getElements();
+        return isLeftOverElement;
+    }
 
-        Map<Integer, List<CsmElement>> elementsToAddBeforeGivenOriginalCSMElement = new HashMap<>();
-        for (int ni = 0; ni< nextOrderElements.size(); ni++) {
-            // If it has a mapping, then it is kept
-            if (!correspondanceBetweenNextOrderAndPreviousOrder.containsKey(ni)) {
-                // Ok, it is something new. Where to put it? Let's see what is the first following
-                // element that has a mapping
-                int originalCsmIndex = -1;
-                for (int nj = ni + 1; nj< nextOrderElements.size() && originalCsmIndex==-1; nj++) {
-                    if (correspondanceBetweenNextOrderAndPreviousOrder.containsKey(nj)) {
-                        originalCsmIndex = correspondanceBetweenNextOrderAndPreviousOrder.get(nj);
-                        if (!elementsToAddBeforeGivenOriginalCSMElement.containsKey(originalCsmIndex)){
-                            elementsToAddBeforeGivenOriginalCSMElement.put(originalCsmIndex, new LinkedList<>());
-                        }
-                        elementsToAddBeforeGivenOriginalCSMElement.get(originalCsmIndex).add(nextOrderElements.get(ni));
+    private void extractReshuffledDiffElements(List<DifferenceElement> diffElements) {
```

Comment:
This seems as big as the methods I use to write :D

Reply:
Well it was just a renaming for the most part :smile:
I didn't really change much here as far as I remember, so....I didn't write it 😝